### DISCUSSION TITLE
Does Forecast Handle IGMP?

### DISCUSSION THREAD

**Discussion by michaelinoue** [strongswan/strongswan#2332] [Q&A] [answered]
For the forecast plugin, Is a separate IGMP daemon necessary for handling multicast group management, or is that taken care of by forecast?

**Comment by tobiasbrunner** [comment, marked as answer]
As [documented](https://docs.strongswan.org/docs/5.9/plugins/forecast.html), the plugin joins the configured groups locally to receive multicast messages to forward.

**Reply by michaelinoue** [reply]
I suppose my question was rather whether it can handle IGMP join/leave/query messages, e.g. adding/removing hosts to/from 
groups dynamically. But it seems that forecast requires a static configuration, so the group management needs to be handled separately.

**Reply by tobiasbrunner** [reply]
While it does forward IGMP messages sent to 224.0.0.1 and 224.0.0.22 (interestingly not 224.0.0.2 by default), it does not analyze them to learn about groups clients join, i.e. it does no IGMP snooping if that's what you were asking.
